### DISCUSSION TITLE
Question about development environment with django-tailwind-cli library

### DISCUSSION THREAD

**Discussion by lucabfelix** [oliverandrich/django-tailwind-cli#138] [General]
Hello, sorry for my lack of experience.

I would like to know if it is advisable for my repository on github to put the assets/css folder and the tailwind.config.js file in .gitignore. Because when I prepare my local environment I will start **python manage.py tailwind runserver** to download the tailwind css. I say this because if I already have my HTML pages styled with Tailwind, when I install the library they will work.

**Comment by oliverandrich** [comment]
My take is, that I put assets/css or specifically the assets/css/tailwind.css in the .gitignore file. During development it is rebuilt all the time. For production it is built while I create my docker container.

The tailwind.config.js is a different story. It is only created once, when it is missing. Afterwards it is part of your source code. Inside you can extend tailwind with it, so you want to have this under source control.

**Reply by lucabfelix** [reply]
Thank you very much for the explanation.

**Comment by lucabfelix** [comment]
Taking advantage of this, I would like to know if it is possible to use the lib without docker. My idea to start is to deploy my system on Digital Ocean for simplicity in deployment.

https://www.digitalocean.com/community/tutorials/how-to-deploy-django-to-app-platform

**Reply by oliverandrich** [reply]
If you want to go this direction, I would also put the compiled tailwind.css file under version control. Then it is always available if you deploy it to app platform or other means of deployment without a docker container.

Personally, I would also add a pre-commit hook, that always takes care of an up-to-date tailwind.css file. A hook that runs the `python manage.py tailwind build` command would be a good idea. Otherwise you might break the app, because you added changes to the html without running a debug server.

**Reply by oliverandrich** [reply]
Maybe it is a good idea to add a best practice section to the documentation.

**Reply by lucabfelix** [reply]
It would be a clever idea to add these practices to the documentation. Most deployments are without Docker. I myself was hesitant to use the lib due to my lack of experience with Docker and being a beginner.

**Reply by oliverandrich** [reply]
Thanks, this is valuable input. The lib should be easy to use and easy to deploy. I will make up my mind on the documentation. I already created the issue #139 for it. I think, I will look into examples for Docker, digital ocean, fly.io and bare-bones deployment to a VPS.

**Reply by lucabfelix** [reply]
It was just a sketch, I'm waiting for news from issue [#139.](https://github.com/django-commons/django-tailwind-cli/issues/139)

Would these be the recommendations you gave me?

**assets/css/tailwind.css with python manage.py tailwind runserver** 
 
**Open the pre-commit hook file:**
nano .git/hooks/pre-commit

**Add the build command:**
python manage.py tailwind build
git add assets/css/tailwind.css

**Give permission to the script:**
chmod +x .git/hooks/pre-commit
